code/game/objects/items/weapons/melee/misc.dm

```diff
@@ -106,3 +106,196 @@
+	///the cooldown and power of enchantments are multiplied by this var when its applied
+	var/power = 1
+
+/obj/item/melee/spellblade/Destroy()
+	QDEL_NULL(enchant)
+	return ..()
+
+/obj/item/melee/spellblade/afterattack(atom/target, mob/user, proximity, params)
+	. = ..()
+	enchant?.on_hit(target, user, proximity, src)
+
+/obj/item/melee/spellblade/attack_self(mob/user)
+	if(enchant)
+		return
+
+	var/list/options = list("Lightning",/// todo add icons for these
+							"Fire",
+							"Bluespace",
+							"Forcewall",)
+	var/list/options_to_type = list("Lightning" = /datum/enchantment/lightning,
+									"Fire" = /datum/enchantment/fire,
+									"Bluespace" = /datum/enchantment/bluespace,
+									"Forcewall" = /datum/enchantment/forcewall,)
+
+	var/choice = show_radial_menu(user, src, options)
+	if(!choice)
+		return
+	add_enchantment(options_to_type[choice], user)
+
+/obj/item/melee/spellblade/proc/add_enchantment(new_enchant, mob/living/user, intentional = TRUE)
+	var/datum/enchantment/E = new new_enchant
+	enchant = E
+	E.on_gain(src, user)
+	E.power *= power
+	if(intentional)
+		SSblackbox.record_feedback("nested tally", "spellblade_enchants", 1, list("[E.name]"))
+
+/obj/item/melee/spellblade/examine(mob/user)
+	. = ..()
+	if(enchant && (iswizard(user) || iscultist(user)))
+		. += "The runes along the side read; [enchant.desc]."
+
+
+/obj/item/melee/spellblade/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(attack_type == PROJECTILE_ATTACK)
+		final_block_chance = 0
+	return ..()
+
+/datum/enchantment
+	/// used for blackbox logging
+	var/name = "You shouldn't be seeing this, file an issue report."
+	/// used for wizards/cultists examining the runes on the blade
+	var/desc = "Someone messed up, file an issue report."
+	/// used for damage values
+	var/power = 0
+	/// whether the enchant procs despite not being in proximity
+	var/ranged = FALSE
+	/// stores the world.time until it can be used again, the `initial(cooldown)` is the cooldown between activations.
+	var/cooldown = -1
+
+/datum/enchantment/proc/on_hit(mob/living/target, mob/living/user, proximity, obj/item/melee/spellblade/S)
+	if(world.time < cooldown)
+		return FALSE
+	if(!istype(target))
+		return FALSE
+	if(target.stat == DEAD)
+		return FALSE
+	if(!ranged && !proximity)
+		return FALSE
+	cooldown = world.time + initial(cooldown)
+	return TRUE
+
+/datum/enchantment/proc/on_gain(obj/item/melee/spellblade, mob/living/user)
+
+/datum/enchantment/lightning
+	name = "lightning"
+	desc = "this blade conducts arcane energy to arc between its victims"
+	power = 20
+	cooldown = 3 SECONDS
+
+/datum/enchantment/lightning/on_hit(mob/living/target, mob/living/user, proximity, obj/item/melee/spellblade/S)
+	. = ..()
+	if(.)
+		zap(target, user, list(user), power)
+
+
+/datum/enchantment/lightning/proc/zap(mob/living/target, mob/living/source, protected_mobs, voltage)
```

Comment:
The list here remains in memory for max 17.5 seconds (if the timers fire as expected). So it shouldn't cause garbage collection issues.